src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionChangeSet.java

```diff
@@ -104,10 +100,16 @@ public NodeOccurrenceCounter getNodesForEdge(final Line line)
         return this.edgeToNodeMapping.get(line.getIdentifier());
     }
 
-    public Set<Long> getPointsThatBecomeNodes()
+    public synchronized Set<Long> getPointsThatBecomeNodes()
```

Comment:
How about a synchronized block? This way it avoids synchronizing on the class. Or is that the only one?

Reply:
This is the only one. Can you explain why having a synchronized block here would be better?

Reply:
With a synchronized block, you can choose what each block synchronizes on. With the synchronized method, it uses the class to synchronize on, meaning that if two or more methods are synchronized, they might wait on each other, even though they would not need to.

Reply:
Good point, I've updated to reflect this.